### ISSUE TITLE
[Other - Accessibility] Instructions likely to be missed by screen reader users. (00.00.1)

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#70908]
#### VFS team name
Veteran Facing Forms

#### VFS product name
Form 40-0247 Presidential Memorial Certificate Form

#### Point of Contact/Reviewers
Brian DeConinck - @briandeconinck - Accessibility

*For more information on how to interpret this ticket, please refer to the [Anatomy of a Staging Review issue ticket](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/Anatomy-of-a-Staging-Review-Issue-ticket.2060320997.html) guidance on Platform Website.

---

### Platform Issue
No VA.gov Experience Standard for the issue found.

### Issue Details
On [step 3](https://staging.va.gov/burials-memorials/memorial-items/presidential-memorial-certificates/request-certificate-form-40-0247/veteran-supporting-documentation), the helper text for file upload isn't announced by screen readers when tabbing to the upload button. It looks like the [solution the team decided on](https://github.com/department-of-veterans-affairs/VA.gov-team-forms/issues/766) was to put some of the text inside the `legend`, which I think makes sense. But with what's in staging all I see in the `legend` is the H3.

### VA.gov Experience Standard
[Category Number 00, Issue Number 00](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/VA.gov-experience-standards.1683980311.html)

### Platform Recommendation
Include the most critical helper text inside the `legend`.

---

### VFS Guidance

- Close the ticket when the issue has been resolved or validated by your Product Owner
- If your team has additional questions or needs Platform help validating the issue, please comment on the ticket
- Some feedback provided may be out of scope for your iteration of the product, however, Platform's OCTO leadership has stated that all identified issues need to be documented and it is still your responsibility to resolve the issue.
- If you do not believe that this Staging Review issue ticket is the responsibility of your team, comment below providing an explanation and who you believe is responsible. Please tag the Point of Contact/Reviewers. Governance team will research and will follow up.

**Comment by tlei123** [comment]
> what's in staging all I see in the `legend` is the H3

@jeana-adhoc , I suggest we have a sync/huddle with Nick and/or Robert to suss this out further.

The suggested fix is already in place but isn't working &mdash; the help-text is indeed inside the `<legend/>` in Staging \[see screenshot below\], but we have another (nested) `<legend/>` rendered by the FileField component.

Furthermore, whatever solution we come up with, if the fix involves the button element itself, then it'd have to be a DST fix because it's inside their `<va-button/>` component.

![image](https://github.com/department-of-veterans-affairs/va.gov-team/assets/587583/f935fc05-43a9-4acf-b7e2-cb2a8744c1a6)

cc @rhasselle-oddball @ndsprinkle

**Comment by jeana-adhoc** [comment]
@tlei123 - I think the issue here is that there's a nested fieldset and nested fieldsets don't work reliably with screen readers. I think we might need to reconsider and use the [file-input component](https://design.va.gov/storybook/?path=/docs/components-va-file-input--default). But then that also creates additional issues. (In the current component pattern , there is a label for the button, and I'm not understanding why that's necessary. And in the version we're using, this same 'label' is in a legend and I'm not sure what purpose its serving either.)

I suggest for now, I work with Rachel, to get a more ideal design set up in sketch and then we can look at it in light of the file input component, and then file the appropriate bugs with the design system.

**Comment by jeana-adhoc** [comment]
@tlei123 - If we're already implementing a custom solution, could we  place the upload button in the same fieldset as the current helper text & H3, and not worry about the "upload documents' text that's in the legend, and remove that completely?  This would remove the nested fieldset.

**Comment by tlei123** [comment]
> If we're already implementing a custom solution, could we place the upload button in the same fieldset as the current helper text & H3, and not worry about the "upload documents' text that's in the legend, and remove that completely? This would remove the nested fieldset.

Yeah, @jeana-adhoc !  I'm gonna give this a try.

**Comment by tlei123** [comment]
> Yeah, @jeana-adhoc ! I'm gonna give this a try. cc @rachelshearerux

@jeana-adhoc , I think it'd be prudent to also start researching now how to fix from the DST-components side...

So far my research indicates that the array pattern _requires_ nested fieldsets &mdash; the outer one for the array & the inner one for each item in the array.

Let me know if you'd like to sync on this.

**Comment by tlei123** [comment]
@jeana-adhoc , @rhasselle-oddball 's suggestion worked!  [PR](https://github.com/department-of-veterans-affairs/vets-website/pull/27225) opened.

**Comment by tlei123** [comment]
[PR](https://github.com/department-of-veterans-affairs/vets-website/pull/27225) merged &mdash; changes should be in Staging by ~4:45 PM.

**Comment by tlei123** [comment]
@jeana-adhoc @rachelshearerux , changes are now [in Staging](https://staging.va.gov/burials-memorials/memorial-items/presidential-memorial-certificates/request-certificate-form-40-0247/), ready for validation.  cc @tbaker1026

**Comment by jeana-adhoc** [comment]
Confirmed in staging!
